ONSdigital/eq-author-app
# 🛠 EAR 1410 - Update schema for `expression.left`

### What is the context of this PR?

Older questionnaires on AWS had validation errors when attempting to reference answers on the same page as routing rules, see e.g. https://author.ons.gov.uk/#/q/28/page/379/routing

Older questionnaires were storing the left hand side of expressions differently to the current version of Author, including `id`s which were identical to the target question (and hence over-writing the position of the original answer in the helper functions which calculate positions within the questionnaire JSON).

This PR adds a migration to remove the fields from older questionnaires which are no longer present in expression's left hand sides.

### How to review

- Import the above Q into local Author
- Check validation error no longer appears for routing logic
- Check that if you move / delete answers the validation errors for routing answers still appear as intended

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
